code/modules/wizardspells/summon_spells.dm

```diff
@@ -300,13 +300,17 @@
 	return 1
 
 /obj/effect/knowspell/summon/world/before_cast(var/mob/caster)
-	targets = list()
-	for(var/mob/living/carbon/human/H in living_mob_list)
-		if(H.stat & DEAD)
-			continue
-		targets += H
-	incant(caster)
-	return 1
+	if(!(src in caster.contents))
```

Comment:
I could be convinced that you should just turn it off for casting from scrolls without CAST_ACTIVATE on the spell.  It would only take a couple lines to fix it that way, probably in the spell topic and scroll description.  That is not a requirement, but another option.